guide/src/guide/new-cell.md

```diff
@@ -1 +1,221 @@
-# Write a New Cell (TODO)
+# Write a New Cell
+
+Following this guide, you can write a custom cell yourself.
+
+## Basic Structure
+
+We have some necessary structs and traits which you must impl for your own cell. We take a cell which loses one packet in every `n` packets (`n` can be set by users freely) as example to explain basic structure of a custom cell.
+
+- `CellIngress`
```

Comment:
We should construct the outline with the name of the traits. 
We should write our guide around the concept of `trait objects`.
You can make them as 3rd level title:

```txt
### Ingress
```

As for `Ingress`, I can give you some example sentences for reference:

---

We should implement an `Ingress` to receive packets, which are dequeued by other cells and forwarded by the rattan core. The `Ingress` serves as the entry of our cell. Since we allow multiple cells to send packets to a single cell from different threads, `Ingress` must implement the `Clone` trait and can only have an immutable reference to itself. However, it's important to note that mutable operations on the packets are possible within the `enqueue` method of `Ingress`.